src/stream/src/executor/dispatch.rs

```diff
@@ -818,6 +862,126 @@ impl Dispatcher for BroadcastDispatcher {
     }
 }
 
+/// Dispatch stream chunk based on table name from upstream DB
+#[derive(Debug)]
+pub struct CdcTableNameDispatcher {
+    outputs: Vec<BoxedOutput>,
+    // column index to the `_rw_table_name` column
+    table_name_col_index: usize,
+    output_indices: Vec<usize>,
+    dispatcher_id: DispatcherId,
+    dispatcher_id_str: String,
+    downstream_table_name: Option<String>,
+}
+
+impl CdcTableNameDispatcher {
+    pub fn new(
+        outputs: Vec<BoxedOutput>,
+        table_name_col_index: usize,
+        output_indices: Vec<usize>,
+        dispatcher_id: DispatcherId,
+        downstream_table_name: Option<String>,
+    ) -> Self {
+        Self {
+            outputs,
+            table_name_col_index,
+            output_indices,
+            dispatcher_id,
+            dispatcher_id_str: dispatcher_id.to_string(),
+            downstream_table_name,
+        }
+    }
+}
+
+impl Dispatcher for CdcTableNameDispatcher {
+    async fn dispatch_data(&mut self, chunk: StreamChunk) -> StreamResult<()> {
+        let num_outputs = self.outputs.len();
+
+        let mut vis_maps = repeat_with(|| BitmapBuilder::with_capacity(chunk.capacity()))
+            .take(num_outputs)
+            .collect_vec();
+
+        let chunk = chunk.project(&self.output_indices);
+
+        for (visible, row) in chunk
+            .visibility()
+            .iter()
+            .zip_eq_fast(chunk.data_chunk().rows_with_holes())
+        {
+            // Build visibility map for every output chunk.
+            for vis_map in &mut vis_maps {
+                let should_emit = if let Some(row) = row && let Some(full_table_name) = self.downstream_table_name.as_ref() {
+                    let table_name_datum = row.datum_at(self.table_name_col_index).unwrap();
+                    tracing::trace!(target: "events::stream::dispatch::hash::cdc", "keys: {:?}, table: {}", self.table_name_col_index, full_table_name);
+                    // dispatch based on downstream table name
+                    table_name_datum == ScalarRefImpl::Utf8(full_table_name)
+                } else {
+                    true
+                };
```

Comment:
Row-level comparison may not be that efficient. We may directly operate on the `full_table_name` column as an `Array`.

Reply:
Since we're going to migrate to the `Filter` executor approach soon, may ignore this.